tests/unit/planout.spec.ts

```diff
@@ -0,0 +1,221 @@
+
+import PlanoutVuePlugin, { PlanoutPlugin } from '@/plugins/planout';
+
+const defaultScript: PlanoutCode = {
+  op: 'seq',
+  seq: [
+    {
+      op: 'set',
+      var: 'variable1',
+      value: 'ns-default',
+    },
+  ],
+};
+
+describe('Vue plugin and vuex module tests', (): void => {
+  const localVue = createLocalVue();
+  const axiosGetMock = jest.fn();
+
+  beforeEach(() => {
+    localVue.use(PlanoutVuePlugin, {
+      appName: 'dummy',
+      baseUrl: 'dummy.com',
+      getConfig: jest.fn(),
+      setConfig: jest.fn(),
+      logExposure: jest.fn(),
+    });
+  });
+
+  // TODO(ram): Make test wait for config load.
+  xit('adds a $planout method to the Vue prototype', async () => {
+    axiosGetMock.mockReturnValueOnce({ namespaces: [], experiments: [] });
+    localVue.use(PlanoutVuePlugin);
+    expect(typeof localVue.prototype.$planout).toBe('object');
+  });
+});
+
+describe('Experiment assignment tests', (): void => {
+  let planout: PlanoutPlugin;
+
+  const simpleScript: PlanoutCode = {
+    op: 'seq',
+    seq: [
+      {
+        op: 'set',
+        var: 'variable1',
+        value: {
+          op: 'uniformChoice',
+          choices: ['goku', 'vegeta'],
+          unit: {
+            op: 'get',
+            var: 'uid',
+          },
+        },
+      },
+    ],
+  };
+
+  const simpleScript2: PlanoutCode = {
+    op: 'seq',
+    seq: [
+      {
+        op: 'set',
+        var: 'variable1',
+        value: {
+          op: 'uniformChoice',
+          choices: ['frieza', 'cell'],
+          unit: {
+            op: 'get',
+            var: 'uid',
+          },
+        },
+      },
+    ],
+  };
+
+  const addExperiment = (namespace: string, name: string, numSegments: number, compiled: PlanoutCode) => {
+    planout.add({
+      namespace_name: namespace,
+      name: name,
+      num_segments: numSegments,
+      compiled: compiled,
+    });
+  };
+
+  let exposures: number;
+  const logExposure = () => {
+    exposures += 1;
+  };
+
+  beforeEach(() => {
+    exposures = 0;
+    planout = new PlanoutPlugin();
+    planout.setOptions({
+      appName: 'dummy',
+      baseUrl: 'dummy.com',
+      getConfig: jest.fn(),
+      setConfig: jest.fn(),
+      logExposure,
+    });
+  });
+
+  it('can run an experiment on all segments', (): void => {
+    addExperiment('ns1', 'exp1', 1000, simpleScript);
+
+    planout.setInput('uid', '1');
+    const value = planout.get('ns1', 'variable1', 'gohan');
+    expect(value).not.toEqual('gohan');
+    expect(['goku', 'vegeta']).toContain(value);
+    expect(exposures).toEqual(1);
+
+    const counts: { [key: string]: number } = {};
+    for (let i = 0; i < 100; i++) {
+      planout.setInput('uid', i.toString());
+      const value = planout.get('ns1', 'variable1', 'gohan');
+      if (!counts[value as string]) {
+        counts[value as string] = 0;
+      }
+      counts[value as string] += 1;
+    }
+    expect(counts.goku).toBeGreaterThan(45);
```

Comment:
Is this process deterministic or are we relying on probability here?

Reply:
It's deterministic

Reply:
ah got it. i think it's worth a comment or a more specific assertion